qiime2/metadata/metadata.py

```diff
@@ -85,7 +87,123 @@ def artifacts(self):
             Source artifacts of the metadata.
 
         """
-        return tuple(self._artifacts)
+        artifacts = []
+        for source in self._sources:
+            if isinstance(source, qiime2.Artifact):
+                artifacts.append(source)
+        return tuple(artifacts)
+
+    # This is to be gotten rid of
+    @property
+    def non_artifacts(self):
```

Comment:
Do we need this property now that we have our "partitioned" sources?

Reply:
I've gone back and forth on that, initially I had it commented "This is to be gotten rid of" up there, I left it because some other things were dependent on it, and I didn't have time to change that yesterday, but now I feel like it probably should be there. It certainly doesn't hurt anything by being there

Reply:
One thing to consider is public API for these objects --- once you set something loose in the wild it can be complicated changing or recalling it. I am going to write up a more detailed comment later today with some skeleton code so that we can discuss.